### ISSUE TITLE
why does ExtensionTotal itself have a "Med Risk (4.80)"?

### ISSUE THREAD

**Issue by EkwzY9PwoKqa7FnPGga6xDigKst3jGmobXnyLAF** [sand-security/extensiontotal-vscode#8]
![image](https://github.com/sand-security/extensiontotal-vscode/assets/168583679/2b97633c-feb3-4242-815d-ec165ed5ffd7)

**Comment by amitassaraf** [comment]
@EkwzY9PwoKqa7FnPGga6xDigKst3jGmobXnyLAF Yes this is because Microsoft is not verifying our domain, possibly because of the research we posted about the downfalls of their systems. We are working on it.

**Comment by EkwzY9PwoKqa7FnPGga6xDigKst3jGmobXnyLAF** [comment]
> @EkwzY9PwoKqa7FnPGga6xDigKst3jGmobXnyLAF Yes this is because Microsoft is not verifying our domain, possibly because of the research we posted about the downfalls of their systems. We are working on it.

thanks for the update. sounds a _little_ conspiratorial but I'm glad you're working on it

**Comment by marknuzz** [comment]
I'm not associated with Microsoft but the pricing model and lack of configuration options to limit API usage is suspicious to me. ~~Until this is addressed I would say it is working as intended.~~ (Edit: appears to be fixed) https://github.com/sand-security/extensiontotal-vscode/issues/10#issuecomment-2197750132

And this is directly relevant to the lack of domain verification. Microsoft could easily name that as a reason for not verifying the domain and it would be hard to argue with them. I do want to see this extension succeed, but it's hard to see how this doesn't look like perverse incentives which undermine the credibility of the extension.

**Comment by EkwzY9PwoKqa7FnPGga6xDigKst3jGmobXnyLAF** [comment]
> I'm not associated with Microsoft but the pricing model and lack of configuration options to limit API usage is suspicious to me. ~Until this is addressed I would say it is working as intended.~ (Edit: appears to be fixed) [#10 (comment)](https://github.com/sand-security/extensiontotal-vscode/issues/10#issuecomment-2197750132)
> 
> And this is directly relevant to the lack of domain verification. Microsoft could easily name that as a reason for not verifying the domain and it would be hard to argue with them. I do want to see this extension succeed, but it's hard to see how this doesn't look like perverse incentives which undermine the credibility of the extension.

Do you think the recent updates improve the extension's credibility?

> I am very happy to update you that due to an increase in organizations adopting ExtensionTotal, ExtensionTotal is now free for personal use! ExtensionTotal's goal has always been to protect as many developers as possible, and I am super thankful for all the support from organizations who has adopted ExtensionTotal.
> 
> So what changed?
> 
> 👉 Individuals who use our [VSCode Extension](http://email.extensiontotal.com/e/c/eyJlbWFpbF9pZCI6ImRnU0c4QWtEQUlFRmdBVUJrRzJYR19kbjRWOUlVdVhWX3VqaSIsImhyZWYiOiJodHRwczovL21hcmtldHBsYWNlLnZpc3VhbHN0dWRpby5jb20vaXRlbXM_aXRlbU5hbWU9ZXh0ZW5zaW9udG90YWwuZXh0ZW5zaW9udG90YWwtdnNjb2RlIiwiaW50ZXJuYWwiOiI4NmYwMDkwMDgwMDU4MTA1IiwibGlua19pZCI6MjB9/f7648e6b9fc93ede82ebee70278076edaa69ee900fa93c7c880f0b46db4df6c5) can now use it for free, all personal API keys were upgraded up to 250 requests per day (should cover all the VSCode power users I spoke with 😉).
> 
> 👉 Organizations who use the API directly now are required to get an [API key here](http://email.extensiontotal.com/e/c/eyJlbWFpbF9pZCI6ImRnU0c4QWtEQUlFRmdBVUJrRzJYR19kbjRWOUlVdVhWX3VqaSIsImhyZWYiOiJodHRwczovL2J1eW1lYWNvZmZlZS5jb20vZXh0ZW5zaW9udG90YWwuc2VjdXJpdHkvbWVtYmVyc2hpcCIsImludGVybmFsIjoiODZmMDA5MDA4MDA1ODEwNSIsImxpbmtfaWQiOjN9/cce211c5122f65633494ea152506928b8bb69bdd7291ca244c178413818e685d) to ensure ExtensionTotal stays free for individuals.

**Comment by marknuzz** [comment]
Definitely an improvement for sure! Appreciate the fast turnaround. It seems fairly straightforward to verify a domain. I did not see any TXT records on extensiontotal.com though.





partial output of dig:
```
$ dig extensiontotal.com

;; QUESTION SECTION:
;extensiontotal.com.		IN	A

;; ANSWER SECTION:
extensiontotal.com.	300	IN	A	99.83.190.102
extensiontotal.com.	300	IN	A	75.2.70.75

;; WHEN: Tue Jul 02 16:23:59 PDT 2024
```


[Eligible domains](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#eligible-domains)
Eligible domains meet the following criteria:

You must be able to manage the DNS configuration settings and add a TXT record.
It is not a subdomain ({subdomain}.github.io, {subdomain}.contoso.com, or similar).
It must use an HTTPS protocol.
It must be able to respond with an HTTP 200 status to a HEAD request.

**Comment by marknuzz** [comment]
Oh wait, I just noticed the other part that makes sense under step 7: https://code.visualstudio.com/api/working-with-extensions/publishing-extension#verify-a-publisher

> Once your TXT record has been validated, the Marketplace team will review your request and let you know the result within 5 business days. The validation includes, but is not limited to: domain, website and extensions content eligibility, legitimacy, trust and positive reputation. New publishers will need a consistent track record demonstrating these for at least 6 months.


So it looks like at minimum the issue is the publisher is simply too new. I knew there was some kind of reputation check  as that's pretty standard, but I did not read it in verbatim before the above comment. So we have confirmation here at least.

**Comment by marknuzz** [comment]
It is certainly possible that Microsoft could use the negative press internally as a reason for rejecting, even after the 6 months. So my advice is to cross every T, dot every I. If ExtensionTotal does everything with a high caliber of credibility like I suggested, Microsoft would not be wise to do a rejection after the fact. It would be far worse of a PR blunder. 

But I think it is also a mistake to suggest that it was possibly retaliation for the negative research when the website clearly shows a minimum age of 6 months. The author's personal GitHub account at least is very established, so that's a good sign.

**Comment by EkwzY9PwoKqa7FnPGga6xDigKst3jGmobXnyLAF** [comment]
@marknuzz I appreciate all the work you're doing in researching! Thank you
